crates/edr_napi/src/account.rs

```diff
@@ -17,7 +25,17 @@ impl TryFrom<GenesisAccount> for edr_provider::AccountConfig {
     type Error = napi::Error;
 
     fn try_from(value: GenesisAccount) -> Result<Self, Self::Error> {
-        let secret_key = secret_key_from_str(&value.secret_key)
+        static_assertions::assert_not_impl_all!(JsString: Debug, Display, Serialize);
+        // `k256::SecretKey` has `Debug` implementation, but it's opaque (only shows the
+        // type name)
+        static_assertions::assert_not_impl_any!(k256::SecretKey: Display, Serialize);
+
+        let secret_key = value.secret_key.into_utf8()?;
+        // This is the only place in production code where it's allowed to use
+        // `DangerousSecretKeyStr`.
+        #[allow(deprecated)]
```

Comment:
I'm wondering whether `deprecated` signals the correct thing. Would this be a good place to use `unsafe`?

An alternative would be to convert `DangerousSecretKeyStr` into an `OpaqueString` type that contains a private string variable and doesn't implement `Debug` or `Display`. Its stored string can be exposed as `str` with a function (potentially using `unsafe`)

Reply:
The message I was going for is "don't use this". The best solution would be imo to make the function private in `edr_napi`, but then there is duplication would be required with `edr_test_utils`. I agree that deprecated is a bit awkward, but it's closer to the "don't use this" semantics for me than `unsafe` as that one is tied to memory safety in my mind which is not the case here.

Reply:
> unsafe [..] is tied to memory safety in my mind which is not the case here.

That a valid point.

I was hoping for an alternative to `#[deprecated]` that causes a compiler error instead of a warning. I did some research but didn't come up with anything 😔 There is no way to configure a project-specific clippy error, otherwise that would have been nice

Reply:
Yeah a custom clippy error would be perfect. But we're running clippy with `-D warnings`, so using the struct will result in a CI error unless the usage is marked with `#[allow(deprecated)]`